pulpcore/app/serializers/publication.py

```diff
@@ -160,6 +160,24 @@ class Meta(ContentGuardSerializer.Meta):
         fields = ContentGuardSerializer.Meta.fields + ("header_name", "header_value", "jq_filter")
 
 
+class NoContentChangeSinceField(serializers.DateTimeField, serializers.ReadOnlyField):
+    """
+    A field representing a timestamp since when the distributed content has not changed.
+    """
+
+    def get_attribute(self, instance):
+        return instance
+
+    def to_representation(self, distribution):
+        publication = distribution.publication
+        repo_version = distribution.repository_version
+
+        if publication or repo_version and not distribution.SERVE_FROM_PUBLICATION:
```

Comment:
Is that meant like this?
```suggestion
        if publication or (repo_version and not distribution.SERVE_FROM_PUBLICATION):
```
(Is a distribution with SERVE_FROM_PUBLICATION being attached to a repository_version even valid?)

Reply:
I wonder if a SerializerMethod field would have been easier.
Also what is the best way a Plugin can subclass this behaviour with better knowledge?

Reply:
> Is a distribution with SERVE_FROM_PUBLICATION being attached to a repository_version even valid?

Conceptually, it might seem a bit wrong. However, I am sure plugin writers can enable both. At the moment, the handler deals only with a scenario where `(repo_version and not distribution.SERVE_FROM_PUBLICATION)`. So, I followed a similar logic here.

> I wonder if a SerializerMethod field would have been easier.

Not a bad idea. [`SerializerMethodField`](https://www.django-rest-framework.org/api-guide/fields/#serializermethodfield) suits our needs.

> Also what is the best way a Plugin can subclass this behaviour with better knowledge?

I think we should go with the SerializerMethod field and plugins can decide to either override the whole logic or do nothing. I do not see any place where we should be able to inject the additional plugin's logic.

Reply:
> I do not see any place where we should be able to inject the additional plugin's logic.

Sure, me neither. Until that one day...

Honestly, with adding more known cases, we may arrive at a point where a plugin may actually know more than pulpcore. So hard to say for sure.